<?xml version="1.0"?>
<launch>

  <arg name="CAM" default="3"/>
  <arg name="dictionary" default="ARUCO_MIP_36h12"/>
  <arg name="marker_size" default="100"/>

  <node type="aruco2_node" name="aruco2_node$(arg CAM)_2" pkg="aruco2" respawn="false" output="screen">
    <param name="topic_in_image" type="string" value="/genicam_cam$(arg CAM)/image_raw"/>
    <param name="image_camera_info" type="string" value="/genicam_cam$(arg CAM)/camera_info"/>
    <param name="topic_out_odom" type="string" value="/cam/odom"/>
    <param name="topic_out_pixel" type="string" value="/cam/pixel"/>
    <param name="pub_pixel" value="1"/>
    <param name="gui" value="1"/>
    <param name="draw_cube" value="0"/>
    <param name="draw_axis" value="0"/>
    <param name="marker_size" value="$(arg marker_size)"/>
    <!--<param name="dictionary" value="$(find aruco2)/dictionary/armarker_parity.dict"/>-->
    <!--<param name="dictionary" value="ARUCO_MIP_36h12"/>-->
    <param name="dictionary" value="$(arg dictionary)"/>
    <param name="verbose" value="0"/>
    <param name="fps" value="50"/>
    <param name="show_threshhold" value="0"/>
  </node>
</launch>
